doc/sphinx/source/hsa.rst

```diff
@@ -0,0 +1,104 @@
+===
+HSA
+===
+
+Note: pocl's HSA support is currently in experimental stage.
+
+The experimental HSA driver works only with an AMD Kaveri or Carrizo APUs
+using the HSAIL-supported LLVM and Clang. Other than that, you will need
+a recent linux (4.0+) and some software.
+
+Installing prerequisite software
+---------------------------------
+
+1) Install the HSA AMD runtime library
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+  Pre-built binaries can be found here:
+
+  https://github.com/HSAFoundation/HSA-Runtime-AMD
+
+  This usually installs into /opt/hsa. Make sure to read Q&A in README.md, it
+  lists some common issues (like /dev/kfd permissions) and run sample/vector_copy
+  to verify you have a working runtime.
+
+2) Build HSAIL-Tools
+~~~~~~~~~~~~~~~~~~~~~
+
+   `git clone` https://github.com/HSAFoundation/HSAIL-Tools
+
+   In particular, **libHSAIL.a** will be required by LLVM, and **HSAILasm** will be
+   required by pocl.
+
+3) Build & install the LLVM with HSAIL support
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+  `git clone` https://github.com/HSAFoundation/HLC-HSAIL-Development-LLVM/
+
+  Use the branch hsail-stable-3.7; before build, patch it with
+
+      `pocl/tools/patches/llvm-3.7-hsail-branch.patch`
+
+  Build it with a Clang 3.7 (branch release_37) patched with
```

Comment:
Add instructions for getting it:
``cd tools; svn co http://llvm.org/svn/llvm-project/cfe/branches/release_37 clang``